### PR TITLE
MINOR: add Eclipse generated files to toplevel .gitignore

### PR DESCRIPTION
**Pull request by edoardocomar**
Ignoring sibfolders' .gitignore except tests
Ignoring core/cache-main and core/cache-tests

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -43,3 +43,9 @@ tests/venv
 .cache
 
 docs/generated/
+
+/core/cache-main
```
Comment (ewencp) [inline .gitignore]: If there are directories under `core/` that need to be ignored, are there not the same directories under other components (e.g. clients, streams, connect, etc)?

File: .gitignore [hunk 2]
```diff
@@ -43,3 +43,9 @@ tests/venv
...
+/core/cache-tests
+/**/.gitignore
```
Comment (ewencp) [inline .gitignore]: Ignoring `.gitignore` files seems wrong...

File: .gitignore (hunk 1 above)
Comment (edoardocomar) [reply .gitignore]: I found those under core only.

File: .gitignore (hunk 2 above)
Comment (edoardocomar) [reply .gitignore]: @ewencp with the exception of the `.gitignore` unders tests that I excluded from being ignores, the other `.gitignore`s are generated by Eclipse and the result is that a freshly cloned Kafka source tree after the `./gradlew eclipse` command had run is full of files that git tells you to commit, while it'd be nice if the tree was clean.

**Comment by edoardocomar** [comment]
A freshly cloned Kafka source tree after the `./gradlew eclipse` command had run is full of files that git tells you to commit, while it'd be nice if the tree was clean.
This PR is not enough to have a clean tree, the Eclipse projects classpaths also need to be fixed but that is, a set of different issues:
https://issues.apache.org/jira/browse/KAFKA-4604
https://issues.apache.org/jira/browse/KAFKA-4617

File: .gitignore (hunk 1 above)
Comment (edoardocomar) [reply .gitignore]: @ewencp I think these are likely coming from the scala plugin, that's why they do not appear under the other projects

File: .gitignore (hunk 2 above)
Comment (edoardocomar) [reply .gitignore]: I retested this after https://github.com/apache/kafka/pull/2382
and even then with the cleaner eclipse structure there are two eclipse generated  gitignores
```
	core/.gitignore
	streams/.gitignore
```

**Comment by edoardocomar** [comment]
@ewencp I verified that 
```
/core/cache-main
/core/cache-tests
```
are generated by the Scala Eclipse IDE
and I have removed ignoring children `.gitignore` after having tried this latest PR to fix the generation of Eclipse artifacts https://github.com/apache/kafka/pull/2382

File: .gitignore [hunk 3]
```diff
@@ -43,3 +43,6 @@ tests/venv
 .cache
 
 docs/generated/
+
+/core/cache-main
+/core/cache-tests
```
Comment (dhwanikatagade) [inline .gitignore]: Shouldn't these be `/core/.cache-main` and `/core/.cache-tests` ?

**Review by dhwanikatagade — CHANGES_REQUESTED** [review]
On my setup having 
Eclipse Neon 4.6.1 
Scala IDE for Eclipse 4.5.0.v-2_11-201612131030-02830a6	
I see the following archive files created. Could you please check once.
```
/core/.cache-main
/core/.cache-tests
```

File: .gitignore (hunk 3 above)
Comment (edoardocomar) [reply .gitignore]: you're right, my oops thanks

**Comment by guozhangwang** [comment]
Is it subsumed by #2382, or are they tackling different issues?

**Comment by edoardocomar** [comment]
different IMO but #2382 has improved it a lot. however one could also use a general configuration of git to avoid polluting .gitignore with IDE artifacts so I'm closing this one

**Pull request closed without merging**